### PR TITLE
PREAPPS-825: Automatic send receive of emails

### PR DESCRIPTION
**Pull request by pl12133**
Return a Promise from the `retry` method resolving after the queue is processed. This allows a consumer to respond to the sync operation beginning/ending.

### CODE REVIEW

File: src/apollo/offline-queue-link/index.ts [hunk 1]
```diff
@@ -155,31 +160,49 @@ export class OfflineQueueLink extends ApolloLink {
...
-						);
-						observer.error(err);
-					}
-				},
-				complete: () => {
-					this.dequeue(entry);
-					if (observer.complete) {
-						observer.complete();
-					}
+	// Retry queries made while offline like apollo-link-queue
```
Comment (Sigafoos) [inline src/apollo/offline-queue-link/index.ts]: ("Actually, everybody likes a pedant" voice) Can you add a period to the end of this line?

File: src/schema/schema.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ export function createZimbraSchema(
 				accountInfo: client.accountInfo,
 				autoComplete: (_, variables) =>
 					client.autoComplete(variables as AutoCompleteOptions),
+				downloadMessage: (_, variables) => client.downloadMessage(variables),
```
Comment (silentsakky) [inline src/schema/schema.ts]: looks like smime related changes are showing up here, which shouldn't happen

Comment (jkester1986) [reply src/schema/schema.ts]: yeah, I was noticing a lot of extra stuff in this PR too

Comment (pl12133) [reply src/schema/schema.ts]: Should be resolved now, I needed to merge those commits into my base branch (PREAPPS-824).

**Pull request merged**